xc7/primitives/idelaye2/idelaye2.pb_type.xml

```diff
@@ -0,0 +1,85 @@
+<!-- set: ai sw=1 ts=1 sta et -->
+<pb_type name="IDELAYE2" num_pb="1">
+  <input name="C" num_pins="1"/>
+  <input name="CE" num_pins="1"/>
+  <input name="CINVCTRL" num_pins="1"/>
+  <input name="CNTVALUEIN0" num_pins="1"/>
+  <input name="CNTVALUEIN1" num_pins="1"/>
+  <input name="CNTVALUEIN2" num_pins="1"/>
+  <input name="CNTVALUEIN3" num_pins="1"/>
+  <input name="CNTVALUEIN4" num_pins="1"/>
+  <input name="DATAIN" num_pins="1"/>
+  <input name="IDATAIN" num_pins="1"/>
+  <input name="IFDLY0" num_pins="1"/>
+  <input name="IFDLY1" num_pins="1"/>
+  <input name="IFDLY2" num_pins="1"/>
+  <input name="INC" num_pins="1"/>
+  <input name="LD" num_pins="1"/>
+  <input name="LDPIPEEN" num_pins="1"/>
+  <input name="REGRST" num_pins="1"/>
+  <output name="CNTVALUEOUT0" num_pins="1"/>
+  <output name="CNTVALUEOUT1" num_pins="1"/>
+  <output name="CNTVALUEOUT2" num_pins="1"/>
+  <output name="CNTVALUEOUT3" num_pins="1"/>
+  <output name="CNTVALUEOUT4" num_pins="1"/>
+  <output name="DATAOUT" num_pins="1"/>
+  <pb_type name="IDELAYE2_BEL" blif_model=".subckt IDELAYE2_BEL" num_pb="1">
+    <clock name="C" num_pins="1"/>
+    <input name="CE" num_pins="1"/>
+    <input name="CINVCTRL" num_pins="1"/>
+    <input name="CNTVALUEIN0" num_pins="1"/>
+    <input name="CNTVALUEIN1" num_pins="1"/>
+    <input name="CNTVALUEIN2" num_pins="1"/>
+    <input name="CNTVALUEIN3" num_pins="1"/>
+    <input name="CNTVALUEIN4" num_pins="1"/>
+    <input name="DATAIN" num_pins="1"/>
+    <input name="IDATAIN" num_pins="1"/>
+    <input name="INC" num_pins="1"/>
+    <input name="LD" num_pins="1"/>
+    <input name="LDPIPEEN" num_pins="1"/>
+    <input name="REGRST" num_pins="1"/>
+    <output name="CNTVALUEOUT0" num_pins="1"/>
+    <output name="CNTVALUEOUT1" num_pins="1"/>
+    <output name="CNTVALUEOUT2" num_pins="1"/>
+    <output name="CNTVALUEOUT3" num_pins="1"/>
+    <output name="CNTVALUEOUT4" num_pins="1"/>
+    <output name="DATAOUT" num_pins="1"/>
+
+    <!-- fake timing values -->
+    <T_setup value="10e-12" port="CE" clock="C" />
+    <T_hold value="10e-12" port="CE" clock="C" />
+    <T_setup value="10e-12" port="INC" clock="C" />
+    <T_hold value="10e-12" port="INC" clock="C" />
+    <T_setup value="10e-12" port="LD" clock="C" />
+    <T_hold value="10e-12" port="LD" clock="C" />
+    <T_setup value="10e-12" port="REGRST" clock="C" />
+    <T_hold value="10e-12" port="REGRST" clock="C" />
+
+  </pb_type>
+  <interconnect>
+    <!-- inputs -->
+    <direct name="C" input="IDELAYE2.C" output="IDELAYE2_BEL.C"/>
+    <direct name="CE" input="IDELAYE2.CE" output="IDELAYE2_BEL.CE"/>
+    <direct name="CINVCTRL" input="IDELAYE2.CINVCTRL" output="IDELAYE2_BEL.CINVCTRL"/>
+    <direct name="CNTVALUEIN0" input="IDELAYE2.CNTVALUEIN0" output="IDELAYE2_BEL.CNTVALUEIN0"/>
```

Comment:
Why have the intermediate pb_type at all if everything is just a straight connection?

Reply:
There are a few muxes to invert signals that I haven't modeled yet.

Reply:
> There are a few muxes to invert signals that I haven't modeled yet.

Those aren't modeled in VPR at all?  VPR does not (currently) understand signal inversion.  I actually believe signal inversion is a synthesis behavior, etc.

Reply:
I believe that we should keep this pb_type as a placeholder for IDELAYE2 which is going to be added there pretty soon.

As for the inverters, I suggest to make the IDELAYE2_VPR pb_type without them (can't model them in VPR anyways). And then simpy invert those signals during techmapping of Vivado IDELAYE2 to VPR's IDELAYE2_VPR according to the primitive parameters.